lib/Sema/DerivedConformanceRawRepresentable.cpp

```diff
@@ -90,6 +90,27 @@ static void deriveBodyRawRepresentable_raw(AbstractFunctionDecl *toRawDecl) {
   }
 #endif
 
+  if (enumDecl->isObjC()) {
```

Comment:
This is a good optimization but isn't safe to perform on types which override `rawValue` for some reason, right?

```swift
@objc enum Foo : Int {
    case a
    
    var rawValue: Int {
        return 5
    }
}

let a = Foo.a
print(a.rawValue) // 5
print(unsafeBitCast(a, to: Int.self)) // 0
```

We should probably see if `rawValue` was synthesized, and if so, bypass calling it with this, yeah?

Reply:
This is the derived implementation of `rawValue`; it'll only be used if the user didn't write one.

(If you're talking about the `hash(into:)` change, I suppose you're right, but we're doomed either way in that case.)

Reply:
Sorry, yes! I was looking in the wrong place.

Reply:
`hash(into:)` could use the bitcast approach; isn't that what `rawValue` does?

Reply:
I suppose it could. I liked the semantic implications of this one, but bitcasting is a little more correct. Do you think it's important enough to be worth changing?

Reply:
Having hashing use the bitcast value feels more inherently correct to me, since an `@objc` enum can in fact be of any value of its raw type, and a hash implementation ought to feed all of the significant bits of the value to the hasher.

Reply:
I realized that `hash(into:)` should probably stay consistent with `==`, which has a default implementation for RawRepresentable types, so I'm going to leave this as is.